Sources/ApolloSQLite/SQLiteNormalizedCache.swift

```diff
@@ -117,3 +117,44 @@ extension SQLiteNormalizedCache: NormalizedCache {
     try self.database.clearDatabase(shouldVacuumOnClear: self.shouldVacuumOnClear)
   }
 }
+
+extension String {
+  private var isBalanced: Bool {
+    guard contains("(") || contains(")") else { return true }
+
+    var stack = [Character]()
+    for character in self where ["(", ")"].contains(character) {
+      if character == "(" {
+        stack.append(character)
+      } else if !stack.isEmpty && character == ")" {
+        _ = stack.popLast()
+      }
+    }
+
+    return stack.isEmpty
+  }
```

Comment:
Maybe this isn't the best name for this since this isn't _strictly_ checking for balanced parentheses pairs.